common/kafka/kafka.go

```diff
@@ -76,33 +78,20 @@ func (sink *kafkaSink) Name() string {
 }
 
 func (sink *kafkaSink) Stop() {
-	// nothing needs to be done.
+	sink.producer.Close()
 }
 
 // setupProducer returns a producer of kafka server
-func setupProducer(sinkBrokerHosts []string, topic string, brokerConf kafka.BrokerConf, compression proto.Compression) (kafka.DistributingProducer, error) {
+func setupProducer(sinkBrokerHosts []string, config *kafka.Config) (kafka.AsyncProducer, error) {
```

Comment:
I do not think this function must exists after the changes.

Reply:
This function is removed and the code is back to NewKafkaClient.